src/common/transformations/src/ngraph_ops/augru_sequence.cpp

```diff
@@ -0,0 +1,180 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "ngraph_ops/augru_sequence.hpp"
+
+#include "itt.hpp"
+#include "ngraph/op/util/recurrent_sequence.hpp"
+
+using namespace std;
+
+BWDCMP_RTTI_DEFINITION(ov::op::internal::AUGRUSequence);
+
+ov::op::internal::AUGRUSequence::AUGRUSequence()
+    : m_direction(op::RecurrentSequenceDirection::FORWARD),
+      m_linear_before_reset(false) {}
+
+ov::op::internal::AUGRUSequence::AUGRUSequence(const Output<Node>& X,
+                                               const Output<Node>& H_t,
+                                               const Output<Node>& sequence_lengths,
+                                               const Output<Node>& W,
+                                               const Output<Node>& R,
+                                               const Output<Node>& B,
+                                               const Output<Node>& A,
+                                               std::size_t hidden_size)
+    : RNNCellBase({X, H_t, sequence_lengths, W, R, B, A},
+                  hidden_size,
+                  0.f,
+                  std::vector<std::string>{"sigmoid", "tanh"},
+                  {},
+                  {}),
+      m_direction(op::RecurrentSequenceDirection::FORWARD),
```

Comment:
it's better to have explicit checks that we support only these values for `m_direction` and `m_linear_before_reset` in validate_and_infer_types(...) method

Reply:
done